plasma_framework/contracts/src/framework/ExitGameController.sol

```diff
@@ -39,89 +41,97 @@ contract ExitGameController is ExitGameRegistry {
         public
         ExitGameRegistry(_minExitPeriod, _initialImmuneExitGames)
     {
-        address ethToken = address(0);
-        exitsQueues[ethToken] = new PriorityQueue();
     }
 
     /**
-     * @notice Add token to the plasma framework and initiate the priority queue.
-     * @notice ETH token is supported by default on deployment.
-     * @dev the queue is created as a new contract instance.
-     * @param _token Address of the token.
+     * @notice Checks if queue for particular token was created.
+     * @param vaultId Id of the vault that handles the token
+     * @param token Address of the token.
+     * @return bool represents whether the queue for a token was created.
      */
-    function addToken(address _token) external {
-        require(!hasToken(_token), "Such token has already been added");
+    function hasExitQueue(uint256 vaultId, address token) public view returns (bool) {
+        bytes32 key = exitQueueKey(vaultId, token);
+        return address(exitsQueues[key]) != address(0);
+    }
 
-        exitsQueues[_token] = new PriorityQueue();
-        emit TokenAdded(_token);
+    function hasExitQueue(bytes32 queueKey) private view returns (bool) {
+        return address(exitsQueues[queueKey]) != address(0);
     }
 
     /**
-     * @notice Checks if the queue for a particular token was created.
-     * @param _token address of the token.
-     * @return bool whether the queue for a token was created.
+     * @notice Adds queue to the plasma framework.
+     * @dev the queue is created as a new contract instance.
+     * @param vaultId Id of the vault
+     * @param token Address of the token.
      */
-    function hasToken(address _token) public view returns (bool) {
-        return address(exitsQueues[_token]) != address(0);
+    function addExitQueue(uint256 vaultId, address token) external {
+        require(vaultId != 0, "Invalid vault id");
```

Comment:
minor: I think the message can be more clear on not able to be 0

Reply:
done